source/isaaclab/isaaclab/managers/event_manager.py

```diff
@@ -387,6 +387,7 @@ def _prepare_terms(self):
 
             # check if the term is a class
             if inspect.isclass(term_cfg.func):
+                term_cfg.func = term_cfg.func(cfg=term_cfg, env=self._env)
```

Comment:
This should be handled by ManagerBase class inside the `_process_term_cfg_at_play` function.